### ISSUE TITLE
Uncaught TypeError: Cannot read property 'id' of undefined

### ISSUE THREAD

**Issue by ybounya** [atom/atom#5702]
[Enter steps to reproduce below:]

1. Select the last line of document
2. hit backspace to delete it
3. Do it again even though there is nothing else to delete on the line

**Atom Version**: 0.181.0
**System**: linux 3.13.0-45-generic
**Thrown From**: Atom Core

### Stack Trace

Uncaught TypeError: Cannot read property 'id' of undefined

```
At /opt/atom/resources/app/node_modules/text-buffer/lib/history.js:124

TypeError: Cannot read property 'id' of undefined
    at TextEditorPresenter.module.exports.TextEditorPresenter.updateLinesState (/opt/atom/resources/app/src/text-editor-presenter.js:246:37)
    at /opt/atom/resources/app/src/text-editor-presenter.js:86:17
    at Emitter.module.exports.Emitter.emit (/opt/atom/resources/app/node_modules/event-kit/lib/emitter.js:82:11)
    at /opt/atom/resources/app/src/text-editor.js:210:32
    at Emitter.module.exports.Emitter.emit (/opt/atom/resources/app/node_modules/event-kit/lib/emitter.js:82:11)
    at DisplayBuffer.module.exports.DisplayBuffer.emitDidChange (/opt/atom/resources/app/src/display-buffer.js:318:20)
    at DisplayBuffer.module.exports.DisplayBuffer.handleBufferMarkersUpdated (/opt/atom/resources/app/src/display-buffer.js:1586:21)
    at /opt/atom/resources/app/src/display-buffer.js:5:61
    at Emitter.module.exports.Emitter.emit (/opt/atom/resources/app/node_modules/event-kit/lib/emitter.js:82:11)
    at TextBuffer.module.exports.TextBuffer.applyPatch (/opt/atom/resources/app/node_modules/text-buffer/lib/text-buffer.js:599:20)
```

### Commands

```
  2x -0:01.0 core:backspace (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {},
  "editor": {
    "invisibles": {},
    "fontSize": 15
  }
}
```

### Installed Packages

```coffee
# User
No installed packages

# Dev
No dev packages
```

**Comment by nathansobo** [comment]
@50Wliu There were a couple distinct issues that both threw this same exception. One of them was fixed in 2e8c5855d406cf939191964a803580cf39b75fd0. The other situation I haven't figured out, but I replaced with a more informative exception in e739444800d09be71cf767caab2ea53a07476055. I'm going to close this for now as I don't expect to see it again, at least in this form. Hope I'm right.

**Comment by karolmie1** [comment]
If you are interested same error message pops up while browsing custom-typings.d.ts file from this generator: http://localhost:3000/#/home

Atom was built from master, d06da3f47046643292c0a451be5ca51c67436a1b
Environment: manjaro linux, vanilla.

> 
> TypeError: Cannot read property 'id' of undefined
>     at TextEditorPresenter.module.exports.TextEditorPresenter.updateLineNumbersState (.../out/Atom/resources/app.asar/src/text-editor-presenter.js:800:56)
>     at TextEditorPresenter.module.exports.TextEditorPresenter.updateTilesState (.../out/Atom/resources/app.asar/src/text-editor-presenter.js:488:14)
>     at TextEditorPresenter.module.exports.TextEditorPresenter.getPreMeasurementState (.../out/Atom/resources/app.asar/src/text-editor-presenter.js:127:12)
>     at TextEditorComponent.module.exports.TextEditorComponent.updateSyncPreMeasurement (.../out/Atom/resources/app.asar/src/text-editor-component.js:265:60)
>     at TextEditorComponent.module.exports.TextEditorComponent.updateSync (.../out/Atom/resources/app.asar/src/text-editor-component.js:205:12)
>     at TextEditorComponent.module.exports.TextEditorComponent.becameVisible (.../out/Atom/resources/app.asar/src/text-editor-component.js:304:21)
>     at TextEditorComponent.module.exports.TextEditorComponent.checkForVisibilityChange (.../out/Atom/resources/app.asar/src/text-editor-component.js:1027:16)
>     at TextEditorComponent.module.exports.TextEditorComponent.pollDOM (.../out/Atom/resources/app.asar/src/text-editor-component.js:1014:17)
>     at .../out/Atom/resources/app.asar/src/text-editor-component.js:3:61
>     at ViewRegistry.module.exports.ViewRegistry.performDocumentPoll (.../out/Atom/resources/app.asar/src/view-registry.js:265:9)
>     at ViewRegistry.module.exports.ViewRegistry.performDocumentUpdate (.../out/Atom/resources/app.asar/src/view-registry.js:228:14)
>     at .../out/Atom/resources/app.asar/src/view-registry.js:3:61
